src/renderer/mod.rs

```diff
@@ -816,7 +820,6 @@ impl<'a> RenderApi<'a> {
             self.render_batch();
         }
     }
-
     pub fn render_cells<I>(
```

Comment:
It looks like the only reason for this method to remain (which shares significant code with the `render_grid` method), is the `render_string` method for rendering the render timer.

I think it should be possible to render text at arbitrary positions without having to use a separate render iterator for it. Potentially it even makes sense to modify the grid itself, similar to the selection? So we could just replace the cells in the RenderableCellsIterator and we would only have to render once, instead of having to render a second time just for the render timer.

Note that for the render timer alone a bit of a hack that renders a second time would be fine, since it's basically never used. But there have been requests for things like searching the scrollback buffer and my PR to display config errors/warnings on startup. These would both profit from having an integrated way to render text at the bottom of the terminal with something like a `render_string` method.

There's no need to go above and beyond with this PR, the main point I was trying to make is that we should refactor the `render_cells` call to make sure we don't have a huge amount of duplicate text. How exactly that is done isn't super important for now. Just calling one from the other should do the job.